File: src/C#/fibonacci.cs
using System;

class Program
{
    static int FibonacciRecursive(int n)
    {
        if (n <= 0) return 0;
        if (n == 1) return 1;
        return FibonacciRecursive(n - 1) + FibonacciRecursive(n - 2);
    }

    static int FibonacciIterative(int n)
    {
        if (n <= 0) return 0;
        if (n == 1) return 1;
        
        int a = 0, b = 1;
        for (int i = 2; i <= n; i++)
        {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }

    static void Main()
    {
        int n = 10;
        int answer1 = FibonacciRecursive(n);
        int answer2 = FibonacciIterative(n);

        Console.WriteLine("Fibonacci recursif (" + n + ") : " + answer1);
        Console.WriteLine("Fibonacci iteratif (" + n + ") : " + answer2);
    }
}

File: src/Rust/fibonacci.rs
fn fibonacci_recursive(n: u32) -> u32 {
    if n == 0 {
        return 0;
    } else if n == 1 {
        return 1;
    } else {
        return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2);
    }
}

fn fibonacci_iterative(n: u32) -> u32 {
    if n == 0 {
        return 0;
    } else if n == 1 {
        return 1;
    }

    let mut a = 0;
    let mut b = 1;
    for _ in 2..n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main(){
    let n = 10;
    let answer1 = fibonacci_recursive(n);
    let answer2 = fibonacci_iterative(n);

    println!("Fibonacci récursif ({}) : {}", n, answer1);
    println!("Fibonacci itératif ({}) : {}", n, answer2);
}